src/main/java/com/coreos/jetcd/EtcdClient.java

```diff
@@ -14,15 +15,22 @@
     private final ManagedChannel channel;
 
     private final EtcdKV kvClient;
+    private final EtcdLease leaseClient;
 
     public EtcdClient(ManagedChannelBuilder<?> channelBuilder, EtcdClientBuilder builder) {
-        this.endpoints = (String[])builder.endpoints().toArray();
-        this.channelBuilder = channelBuilder != null ? channelBuilder : ManagedChannelBuilder.forAddress("localhost", 2379);
+        this.endpoints = new String[builder.endpoints().size()];
+        builder.endpoints().toArray(this.endpoints);
```

Comment:
I encountered a  exception(java.lang.ClassCastException) here without this change in com.coreos.jetcd.EtcdLeaseTest  , =,=